apps/meteor/ee/server/models/raw/ReadReceipts.ts

```diff
@@ -16,4 +17,92 @@ export class ReadReceiptsRaw extends BaseRaw<ReadReceipt> implements IReadReceip
 	findByMessageId(messageId: string): FindCursor<ReadReceipt> {
 		return this.find({ messageId });
 	}
+
+	removeByUserId(userId: string): Promise<DeleteResult> {
+		return this.deleteMany({ userId });
+	}
+
+	removeByRoomId(roomId: string): Promise<DeleteResult> {
+		return this.deleteMany({ roomId });
+	}
+
+	removeByRoomIds(roomIds: string[]): Promise<DeleteResult> {
+		return this.deleteMany({ roomId: { $in: roomIds } });
+	}
+
+	removeByMessageId(messageId: string): Promise<DeleteResult> {
+		return this.deleteMany({ messageId });
+	}
+
+	removeByMessageIds(messageIds: string[]): Promise<DeleteResult> {
+		return this.deleteMany({ messageId: { $in: messageIds } });
+	}
+
+	removeOTRReceiptsUntilDate(roomId: string, until: Date): Promise<DeleteResult> {
+		const query = {
+			roomId,
+			t: {
+				$in: [
+					'otr',
+					otrSystemMessages.USER_JOINED_OTR,
+					otrSystemMessages.USER_REQUESTED_OTR_KEY_REFRESH,
+					otrSystemMessages.USER_KEY_REFRESHED_SUCCESSFULLY,
+				],
+			},
+			ts: { $lte: until },
+		};
+		return this.deleteMany(query);
+	}
+
+	async removeByIdPinnedTimestampLimitAndUsers(
+		roomId: string,
+		ignorePinned: boolean,
+		ignoreDiscussion: boolean,
+		ts: Filter<IMessage>['ts'],
+		users: IUser['_id'][],
+		ignoreThreads: boolean,
+	): Promise<DeleteResult> {
+		const query: Filter<ReadReceipt> = {
+			roomId,
+			ts,
+		};
+
+		if (ignorePinned) {
+			query.pinned = { $ne: true };
+		}
+
+		if (ignoreDiscussion) {
+			query.drid = { $exists: false };
+		}
+
+		if (ignoreThreads) {
+			query.tmid = { $exists: false };
+			query.tcount = { $exists: false };
+		}
+
+		if (users.length) {
+			query.userId = { $in: users };
+		}
+
+		return this.deleteMany(query);
+	}
+
+	setPinnedByMessageId(messageId: string, pinned?: boolean): Promise<Document | UpdateResult> {
+		if (pinned == null) {
```

Comment:
In which cases can you pass null in as `pinned`?

Why cannot use a default value of `true` for pinned param?

Reply:
Done!